### PR TITLE
fix: loosen restrictions on Tabs children

### PR DESCRIPTION
**Pull request by zchsh**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-zsloosen-tabs-validation-hashicorp.vercel.app/vault/tutorials/secrets-management/multi-factor-authentication) 🔎
- [Asana task](https://app.asana.com/0/1202001387788447/1202198050544677/f) 🎟️

## 🗒️ What

Fixes a runtime error related to `Tabs`

## 🤷 Why

Runtime errors are bad.

## 🛠️ How

Loosens `Tabs` child restrictions, to allow it to accept a single child.

## 🧪 Testing

- [ ] Visit [/vault/tutorials/secrets-management/multi-factor-authentication](https://dev-portal-git-zsloosen-tabs-validation-hashicorp.vercel.app/vault/tutorials/secrets-management/multi-factor-authentication), which is the route mentioned in the bug ticket, and ensure it renders as expected

## 💭 Anything else?

Nope! As mentioned in the task, I think ultimately we will want to throw for this issue - but as long as it's possible for authors to push this content through without breaking a build, it seems we'll need to continually loosen restrictions.

### CODE REVIEW

File: src/components/tabs/hooks/use-tab-items.ts [hunk 1]
```diff
@@ -94,16 +94,20 @@ function validateTabChildren(
...
+   * Warn if rendering a `Tabs` component with only one child.
    */
   if (!Array.isArray(children) || children.length === 1) {
-    throw new Error('children must be an array of multiple Tab components')
+    console.warn(
+      'Warning: Tabs children must be an array of multiple Tab components. In the future, use of a single Tab will throw an error.'
+    )
   }
 
+  const childrenArray = Children.toArray(children)
```
Comment (zchsh) [inline src/components/tabs/hooks/use-tab-items.ts:105]: `children` is no longer guaranteed to be an array; we need to make sure it is.

**Comment by zchsh** [comment]
Note: for some reason a different `Tabs` component on the same page is being rendered into what appears to be an `Image` container. Looking into this in a separate ticket: 🎟️ [[mdx bug] tabs rendering in an image container for some reason](https://app.asana.com/0/1202097197789424/1202221422987978/f).

File: src/components/tabs/hooks/use-tab-items.ts [hunk 2]
```diff
@@ -94,16 +94,20 @@ function validateTabChildren(
   initialActiveIndex: number
 ): RawTabItem[] {
   /**
-   * Disallow rendering a `Tabs` component with only one child.
+   * Warn if rendering a `Tabs` component with only one child.
```
Comment (kendallstrautman) [inline src/components/tabs/hooks/use-tab-items.ts:97]: For this case, was it just that one tab rendered a single codeblock? That _seems_ like a reasonable usecase, no?

**Review by kendallstrautman — APPROVED** [review]
Awesome, ty for fixing!

**Pull request merged**